.pre-commit-config.yaml

```diff
@@ -101,3 +101,16 @@ repos:
         exclude: *fixtures
         args: ["--split-summary-body", "--max-summary-lines=2"]
         files: "pylint"
+  - repo: https://github.com/cdce8p/python-typing-update
+    rev: v0.4.1
+    hooks:
+      - id: python-typing-update
+        stages: [manual]
+        additional_dependencies:
+          - "black==22.3.0"
+        args:
+          - --py37-plus
+          - --force
+          - --keep-updates
+          - --black
+        files: ^(pylint)/.+\.py$
```

Comment:
I agree with what Marc said earlier, that pylint's typing extension is vastly more efficient. I would rather add the typing extension to our pylintrc. I think we should only handle the bump of the py-version in this and discuss this in another one.

Reply:
> I would rather add the typing extension to our pylintrc.

It's already enabled. We just need to update `py-version`.

> I think we should only handle the bump of the py-version in this MR and discuss the pre-commit hook in another one.

AFAICT the typing changes here are from #6325.
With the typing extension, we don't need to add any other pre-commit hook 😄